appimagecraft/generators/appimage_build_script.py

```diff
@@ -20,7 +21,10 @@ def __init__(self, ld_config: dict = None):
     def build_file(self, path: str):
         gen = BashScriptGenerator(path)
 
-        arch = self._config.get("arch", platform.machine())
+        if "ARCH" in os.environ:
+            arch = os.environ["ARCH"]
```

Comment:
You should provide a warning that signalizes to the user that data from the environment was taken into account.